addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/discovery/SomfyTahomaItemDiscoveryService.java

```diff
@@ -39,16 +37,67 @@
 public class SomfyTahomaItemDiscoveryService extends AbstractDiscoveryService implements ExtendedDiscoveryService {
 
     private final Logger logger = LoggerFactory.getLogger(SomfyTahomaItemDiscoveryService.class);
-    private SomfyTahomaBridgeHandler bridge = null;
+    private SomfyTahomaBridgeHandler bridge;
     private DiscoveryServiceCallback discoveryServiceCallback;
+    private ScheduledFuture<?> discoveryJob;
+
 
     private static final int DISCOVERY_TIMEOUT_SEC = 10;
+    private static final int DISCOVERY_REFRESH_SEC = 30;
 
     public SomfyTahomaItemDiscoveryService(SomfyTahomaBridgeHandler bridgeHandler) {
         super(DISCOVERY_TIMEOUT_SEC);
         logger.debug("Creating discovery service");
         this.bridge = bridgeHandler;
-        bridgeHandler.setDiscoveryService(this);
+    }
+
+    /**
+     * Called on component activation.
+     */
+    public void activate() {
+        super.activate(new HashMap<>());
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    public void setDiscoveryServiceCallback(DiscoveryServiceCallback discoveryServiceCallback) {
+        this.discoveryServiceCallback = discoveryServiceCallback;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Starting SomfyTahoma background discovery");
+
+        if (discoveryJob == null || discoveryJob.isCancelled()) {
+            try {
+                discoveryJob = scheduler.scheduleWithFixedDelay(this::runDiscovery, 10, DISCOVERY_REFRESH_SEC,
+                        TimeUnit.MINUTES);
+            } catch (java.lang.IncompatibleClassChangeError ex) {
+                logger.warn("This binding is most probably running in an OpenHAB 2.2.x or older environment.");
+                logger.warn("Auto discovery is not compatible, please use manual one.");
+            } catch (Exception ex) {
```

Comment:
What kind of Exceptions do you expect here, please catch the specific ones or remove the try-catch-block.

Reply:
Removed